### PR TITLE
feat: add command for user specified compaction

### PR DESCRIPTION
**Pull request by levy5307**
RFC: https://github.com/apache/incubator-pegasus/blob/master/rfcs/2021-05-27-user-specified-compaction.md

### CODE REVIEW

File: executor/compaction.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+	operations.Ops = append(operations.Ops, *operation)
+	res, _ := json.Marshal(operations)
+	return string(res), nil
+}
+
+func generateUpdateTTLOperation(updateTTLType string, timeValue uint) (*compactionOperation, error) {
+	var params updateTTLParams
+	params.Value = timeValue
+	switch updateTTLType {
```
[the remark this answers is not in the document]
Comment (levy5307) [reply executor/compaction.go:99]: Done

File: executor/table_env.go [hunk 2]
```diff
@@ -41,6 +41,24 @@ func ListAppEnvs(c *Client, useTable string) error {
...
+func GetAppEnv(c *Client, useTable string, envName string) (string, error) {
+	tables, err := c.Meta.ListAvailableApps()
+	if err != nil {
+		return "", err
+	}
+
+	for _, table := range tables {
+		if table.AppName == useTable {
+			for k, v := range table.Envs {
+				if k == envName {
```
[the remark this answers is not in the document]
Comment (levy5307) [reply executor/table_env.go:53]: I will do it in later pull request

File: cmd/compaction.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+				c.UseTable,
+				c.Flags.String("operation-type"),
+				c.Flags.String("ttl-type"),
+				c.Flags.Uint("time-value"),
+				c.Flags.String("hashkey-pattern"),
+				c.Flags.String("hashkey-match"),
+				c.Flags.String("sortkey-pattern"),
+				c.Flags.String("sortkey-match"),
+				c.Flags.Int64("start-ttl"),
+				c.Flags.Int64("stop-ttl"),
```
[the remark this answers is not in the document]
Comment (levy5307) [reply cmd/compaction.go:46]: I think it's ok to use these params

[the remark this answers is not in the document]
Comment (Shuo-Jia) [reply cmd/compaction.go:46]: I still think the params is too many, you could find many function all need pass these params, it's too complex

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**